textdb/textdb-dataflow/src/main/java/edu/uci/ics/textdb/dataflow/queryrewriter/QueryRewriter.java

```diff
@@ -64,7 +79,11 @@ public ITuple getNextTuple() throws Exception {
         if(!isOpen || endOfResult)
             return null;
         else {
-            List<String> queryStrings = QuerySegmenter.getTokens(searchQuery);
+            List<String> queryStrings;
+            if(allSegmentations)
+                queryStrings = QuerySegmenter.bruteGetTokens(searchQuery);
```

Comment:
We were thinking of refactoring `bruteGetTokens` -> `getAllTokens` and `getTokens` -> `getLikelyTokens`.  
`getAllTokens` can internally call `rewriteTermBruteForce` and `getLikelyTokens` can call `rewriteTermDP`  
@chenlica : Do you think that would be ok?